deepchem/models/tf_new_models/tensor_graph.py

```diff
@@ -0,0 +1,238 @@
+      # TODO(LESWING) do this lazily and call in Topological sort order after call to "build"
+      layer.set_parents(parents)
+      return layer
+
+  def fit(self,
+          dataset,
+          nb_epoch=10,
+          max_checkpoints_to_keep=5,
+          log_every_N_batches=50,
+          learning_rate=.001,
+          batch_size=50,
+          checkpoint_interval=10):
+    """Trains the model for a fixed number of epochs.
+
+    TODO(rbharath0: This is mostly copied from TensorflowGraphModel. Should
+    eventually refactor both together.
+
+    Parameters
+    ----------
+    dataset: dc.data.Dataset
+    nb_epoch: 10
+      Number of training epochs.
+      Dataset object holding training data
+        batch_size: integer. Number of samples per gradient update.
+        nb_epoch: integer, the number of epochs to train the model.
+        verbose: 0 for no logging to stdout,
+            1 for progress bar logging, 2 for one log line per epoch.
+        initial_epoch: epoch at which to start training
+            (useful for resuming a previous training run)
+    checkpoint_interval: int
+      Frequency at which to write checkpoints, measured in epochs
+    """
+    with self.graph.as_default():
+      time1 = time.time()
+      print("Training for %d epochs" % nb_epoch)
+      self.train_op = tf.train.AdamOptimizer(learning_rate).minimize(self.loss)
+      with tf.Session() as sess:
+        sess.run(tf.global_variables_initializer())
+        saver = tf.train.Saver(max_to_keep=max_checkpoints_to_keep)
+        # Save an initial checkpoint.
+        saver.save(sess, self.data_dir, global_step=0)
+        for epoch in range(nb_epoch):
+          avg_loss, n_batches = 0., 0
+          # TODO(rbharath): Don't support example weighting yet.
+          for ind, (X_b, y_b, w_b, ids_b) in enumerate(
+              dataset.iterbatches(batch_size, pad_batches=True)):
+            if ind % log_every_N_batches == 0:
+              print("On batch %d" % ind)
+            feed_dict = {self.features: X_b, self.labels: y_b}
+            fetches = [self.outputs] + [self.train_op, self.loss]
+            fetched_values = sess.run(fetches, feed_dict=feed_dict)
+            loss = fetched_values[-1]
+            avg_loss += loss
+            n_batches += 1
+          if epoch % checkpoint_interval == checkpoint_interval - 1:
+            saver.save(sess, self.data_dir, global_step=epoch)
+          avg_loss = float(avg_loss) / n_batches
+          print('Ending epoch %d: Average loss %g' % (epoch, avg_loss))
+          # Always save a final checkpoint when complete.
+          saver.save(sess, self.data_dir, global_step=epoch + 1)
+      ############################################################## TIMING
+      time2 = time.time()
+      print("TIMING: model fitting took %0.3f s" % (time2 - time1))
+      ############################################################## TIMING
+
+
+class Conv1DLayer(Layer):
```

Comment:
We should move these into `dc.nn.layers` or `dc.nn.copy` (`copy` indicates "copied" over from keras. We may just want to merge `dc.nn.layers` and `dc.nn.copy` together)